samplers/aws/xray/rule_manifest.go

```diff
@@ -0,0 +1,231 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xray
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+)
+
+const defaultRule = "Default"
+const defaultInterval = int64(10)
+
+//const manifestTTL = 3600 // Seconds
+
+//// centralizedManifest represents a full sampling ruleset, with a list of
+//// custom rules and default values for incoming requests that do
+//// not match any of the provided rules.
+type centralizedManifest struct {
+	defaultRule *centralizedRule
+	rules       []*centralizedRule
+	index       map[string]*centralizedRule
+	refreshedAt int64
+	clock       Clock
+	mu          sync.RWMutex
+}
+
+// putRule updates the named rule if it already exists or creates it if it does not.
+// May break ordering of the sorted rules array if it creates a new rule.
+func (m *centralizedManifest) putRule(rule *ruleProperties) (r *centralizedRule, err error) {
+	defer func() {
+		if x := recover(); x != nil {
+			err = fmt.Errorf("%v", x)
+		}
+	}()
+
+	name := *rule.RuleName
```

Comment:
```suggestion
	name := rule.RuleName
```

The dereference isn't necessary here, is it?